outbound/hmail.js

```diff
@@ -618,10 +618,12 @@ class HMailItem extends events.EventEmitter {
             else {
                 self.discard();
             }
-            if (cfg.pool_concurrency_max && !mx.using_lmtp) {
+
+            if (cfg.pool_concurrency_max && success) {
+                client_pool.release_client(socket, port, host, mx.bind, fin_sent);
+            } else if (cfg.pool_concurrency_max && !mx.using_lmtp) {
                 send_command('RSET');
-            }
-            else {
+            } else {
```

Comment:
Don't do that.

Reply:
@baudehlo , see https://github.com/haraka/haraka-eslint/pull/16

Reply:
Should be ok by now I guess?

Reply:
Yes, cuddled else's have shown up in several PRs in recent memory, so rather than having humans point it out, we can let eslint do the job.